src/services/users.service.ts

```diff
@@ -28,25 +31,49 @@ export class UsersService {
     user.password = this.hashPassword(user.password);
   }
 
-  listUsers() {
+  listUsers(): Promise<User[]> {
     return this.userRepository.find();
   }
 
-  showUser(id: number) {
-    return this.userRepository.findOne(id);
+  async showUser(id: number): Promise<User> {
+    const user = await this.userRepository.findOne(id);
+    if (!user) {
+      throw new UserNotFoundError( );
+    }
+    return user;
```

Comment:
Awesome implementation! One tiny tip, just to show you another way to do this.

If you look at the TypeORM API you will find "findOne" method, which allows you to retrieve a single entity when a condition is met, but also there is a "findOneOrFail" function which automatically rejects the promise if the requested entity is not found for the criteria you have passed. 

So you could rewrite the same function as this and replace the if implementation for a modern/try-catch

```typescript
async showUser(id: number): Promise<User> {
try{
    const user = await this.userRepository.findOneOrFail(id);
    return user;
} catch(error){
   throw new UserNotFoundError();
}
```

Again, this change is not mandatory, it's just a suggestion to show you there's more than one way to do this kind of stuff.